### PR TITLE
Set default applciations dir path value to $HOME/Applications

### PR DESCRIPTION
**Pull request by azubieta**
partial fix for #218

### CODE REVIEW

**Comment by TheAssassin** [comment]
Well, actually this should be *empty* by default. Think "if it's empty use the default". That way, nothing is hardcoded to the config file. If the user sets something explicitly, that is used then, but if it's empty the default should be used.

I think the code tries to clear that somewhere else. So all this does is overwrite the empty value again with a path. And that is unintended behavior.

**Comment by azubieta** [comment]
> this should be _empty_ by default

That would be extremely uninformative. It could be also interpreted as disabled if the default behavior is to use $HOME/Applications then that should be de default value. If you write it down to the config file or not that's your call.

**Comment by TheAssassin** [comment]
Not writing it _even if it is set_ is more confusing IMO. That would be very implicit behavior. Leaving it empty gives a hint towards "use default".

If you want to put it in there, please develop a mechanism like using gray text that, if changed, gets black, but only if it's changed. I think you don't even need a `bool` for that, it can all be read from the internal config object. That's the UX I initially wanted for this, but I couldn't get it done within the time I had back then. When the user clears it, that text should be shown.
In web development, text fields got some `placeholder` attribute. AFAICS there's nothing like that in Qt. But if there is, this would be *really* easy to improve without giving up on signalizing that "currently uses default value" UX element that I consider *really* important.

Also, really problematic in your code: you did not use the central method to get this path, but hardcoded something. That's IMO not a good idea.

**Comment by azubieta** [comment]
> Also, really problematic in your code: you did not use the central method to get this path, but hardcoded something. That's IMO not a good idea.

Didn't knew about the existence of `integratedAppImagesDestination()` now is being used.

File: src/ui/settings_dialog.cpp [hunk 1]
```diff
@@ -54,7 +54,7 @@ void SettingsDialog::loadSettings() {
         ui->checkBoxAskMove->setChecked(settingsFile->value("AppImageLauncher/ask_to_move", false).toBool());
         ui->lineEditApplicationsDir->setText(settingsFile->value("AppImageLauncher/destination").toString());
     } else {
-        ui->lineEditApplicationsDir->setText(QDir::homePath() + "/Applications");
+        ui->lineEditApplicationsDir->setPlaceholderText(integratedAppImagesDestination().absolutePath());
```
Comment (TheAssassin) [inline src/ui/settings_dialog.cpp]: Maybe I was totally blind, but this looks awesome. Gonna build that quickly to review it.

Comment (TheAssassin) [reply src/ui/settings_dialog.cpp]: So, by default, it looks like this:

screenshot_2019-07-31_23-37-56

I guess this line isn't called during startup...

**Comment by azubieta** [comment]
Just pushed another small improvement.

File: src/ui/settings_dialog.cpp (hunk 1 above)
Comment (azubieta) [reply src/ui/settings_dialog.cpp]: did you pulled the latest changes? This is how it looks to me

Screenshot_20190731_163929

Comment (TheAssassin) [reply src/ui/settings_dialog.cpp]: Already fixed it. Let me review the entire code base once again.

Comment (TheAssassin) [reply src/ui/settings_dialog.cpp]: As it's a placeholder, it should be set anyway. Moving it to the constructor did the trick.

**Pull request merged**